### PR TITLE
Enhancement: Show confirmation modal if a workflow would 'steal' a content-type

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

1. Adds a new prompt that is displayed if assigning a content-type to a workflow would 'steal' this content-type from another workflow
2. Display a notice about which workflows are already assigned in the content-types select
3. Combine the above prompt with the existing prompt of deleted server stages
4. Refactor the `ConfirmDialog` component to be more composable

### Why is it needed?

This will make sure users understand the impact of selecting a specific content-type for a workflow.

### How to test it?

1. Create 2 workflows
5. Assign an already assigned content-type to the other workflow: expect a prompt to be displayed when saving
6. Delete an already existing stage: expect a prompt to be displayed when saving, which contains two questions

### Related issue(s)/PR(s)

- Alternative for https://github.com/strapi/strapi/pull/17392
- Closes https://github.com/strapi/strapi/pull/17392

### CODE REVIEW

**Comment by gu-stav** [comment]
@joshuaellis The tests will probably be broken and I have to add a few new ones. Just wanted to get a quick feedback on whether this approach works for you.

File: packages/core/helper-plugin/src/components/ConfirmDialog/index.js [hunk 1]
```diff
@@ -119,4 +245,7 @@ ConfirmDialog.propTypes = {
   variantRightButton: PropTypes.string,
 };
 
-export default ConfirmDialog;
+ConfirmDialog.Root = Root;
+ConfirmDialog.Body = Body;
+
+export { ConfirmDialog };
```
Comment (joshuaellis) [inline packages/core/helper-plugin/src/components/ConfirmDialog/index.js:251]: Nice

**Comment by joshuaellis** [comment]
Nice work on the modal stuff too 👍🏼

**Review by joshuaellis — DISMISSED** [review]
Noice Thats Nice GIF

**Comment by gu-stav** [comment]
@joshuaellis Sorry - had to fix a smol typo in the confirmation dialog text.

**Review by joshuaellis — APPROVED** [review]
you dont get a gif this time.

**Comment by gu-stav** [comment]
Do I get another one if I keep rebasing?

**Comment by joshuaellis** [comment]
> Do I get another one if I keep rebasing?

no

**Pull request merged**